src/bin/casc/main.rs

```diff
@@ -34,25 +33,58 @@ fn main() -> std::io::Result<()> {
             "No policy source files found",
         ));
     }
-    let mut out_file = File::create(args.out_filename)?;
-    let res = compile_system_policy(policies.iter().map(|s| s as &str).collect());
-    match res {
-        Err(error_list) => {
-            for e in error_list {
-                if let ErrorItem::Parse(p) = e {
-                    p.print_diagnostic();
-                } else if let ErrorItem::Compile(c) = e {
-                    c.print_diagnostic();
-                } else {
-                    eprintln!("{}", e);
+
+    // If no system names are given, output a single CIL file containing all of the policies,
+    // with the default out file name (out.cil) if an out file name isn't specified.
+    // Else, if the system name given is "all", build all of the systems.
+    // This assumes that "all" is a reserved keyword, so a system cannot be declared with the name "all".
+    // Otherwise, output an individual CIL files for each of the system names given.
+    // In both of the previous two cases, the name of each output CIL file is the name of the system + .cil.
+    if args.system_names.is_empty() {
+        let mut out_file = File::create(args.out_filename.clone())?;
+        let res = compile_combined(
+            policies.iter().map(|s| s as &str).collect(),
+            args.out_filename,
+        );
+        match res {
+            Err(error_list) => print_error(error_list),
+            Ok(s) => out_file.write_all(s.as_bytes()),
```

Comment:
Can we unify the writing out code by having the part that calls compile_combined make its own HashMap with the default system name?